bin/truffleruby.sh

```diff
@@ -77,7 +99,14 @@ java_args+=("-Xbootclasspath/a:$truffle/dists/truffle-api.jar:$graal_sdk")
 CP="$CP:$truffle/dists/truffle-nfi.jar"
 CP="$CP:$root/mxbuild/dists/truffleruby-launcher.jar"
 CP="$CP:$root/mxbuild/dists/truffleruby.jar"
-java_args+=("-Dtruffle.nfi.library=$truffle/truffle-nfi-native/bin/libtrufflenfi.$(libext)")
+
+libtrufflenfi_os_arch="$truffle/$os-$arch/truffle-nfi-native/bin/libtrufflenfi.$libext"
+if [ -f "$libtrufflenfi_os_arch" ]; then
+  libtrufflenfi="$libtrufflenfi_os_arch"
+else
+  libtrufflenfi="$truffle/truffle-nfi-native/bin/libtrufflenfi.$libext"
```

Comment:
Please include not-found path https://github.com/graalvm/truffleruby/pull/689/files#diff-851e210aae12b5c6f8feb97ae3ac838aR95 and a todo to remove the fallback https://github.com/graalvm/truffleruby/pull/689/files#diff-851e210aae12b5c6f8feb97ae3ac838aR82

Reply:
I'll add the TODO.
For the not found case, I think the UnsatisifiedLinkError which is caused later is clear enough, but we might as well check earlier, right.

Reply:
Done.